server.go

```diff
@@ -22,36 +21,36 @@ func computeSum(body []byte) []byte {
 	h := sha256.New()
 
 	h.Write(body)
-	hashed := h.Sum(nil)
-	return hashed
+	hashed := hex.EncodeToString(h.Sum(nil))
+	return []byte(hashed)
 }
 
 func hashHandler(histogram *prometheus.HistogramVec) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		start := time.Now()
 		defer r.Body.Close()
-		code := 500
+		code := http.StatusInternalServerError
 
 		defer func() { // Make sure we record a status.
 			duration := time.Since(start)
 			histogram.WithLabelValues(fmt.Sprintf("%d", code)).Observe(duration.Seconds())
 		}()
 
 		code = http.StatusBadRequest
-		if r.Method == "POST" {
-			code = http.StatusOK
+		if r.Method != "POST" {
```

Comment:
If we're doing refactoring for the sake of it then you should also use `http.MethodPost` here..

Reply:
Absolutely I totally missed that one. Updated.